projects/batfish/src/main/java/org/batfish/bdp/VirtualRouter.java

```diff
@@ -217,53 +219,95 @@ public void computeFib() {
     _fib = new Fib(_mainRib);
   }
 
-  public boolean computeInterAreaSummaries() {
+  /**
+   * Decides whether the current OSPF summary route metric needs to be changed based on the given
+   * route's metric.
+   *
+   * <p>Routes from the same area or outside of areaPrefix have no effect on the summary metric.
+   *
+   * @param route The route in question, whose metric is considered
+   * @param areaPrefix The Ip prefix of the OSPF area
+   * @param currentMetric The current summary metric for the area
+   * @param areaNum Area number.
+   * @param useOldRfc Whether to use the older RFC 1583 computation, which takes the minimum of
+   *     metrics as opposed to the newer RFC 2328, which uses the maximum
+   * @return the newly computed summary metric.
+   */
+  @Nullable
+  static Long computeUpdatedOspfSummaryMetric(
+      OspfAreaRoute route,
+      Prefix areaPrefix,
+      @Nullable Long currentMetric,
+      long areaNum,
+      boolean useOldRfc) {
+    Prefix contributingRoutePrefix = route.getNetwork();
+    // Only update metric for different areas and if the area prefix contains the route prefix
+    if (areaNum == route.getArea() || !areaPrefix.containsPrefix(contributingRoutePrefix)) {
+      return currentMetric;
+    }
+    long contributingRouteMetric = route.getMetric();
+    // Definitely update if we have no previous metric
+    if (currentMetric == null) {
+      return contributingRouteMetric;
+    }
+    // Take the best metric between the route's and current available
+    /*
+     * NOTE: Best was determined using the min function according to RFC 1583.
+     * However, OSPFv2 uses max function.
+     * Cisco claims to have switched to using max in IOS v.12.0 and later,
+     * as described in RFC 2328.
+     * (see https://www.cisco.com/c/en/us/support/docs/ip/open-shortest-path-first-ospf/7039-1.html#t29)
```

Comment:
Cisco apparently defaults to "compatible rfc1583", so `min` is correct here. Please add a TODO here (and file a bug) for adding parsing support to handle `no compatible rfc1583` and plumbing it through to the dataplane (`useOldRfc` parameter on this function).

Also suggest renaming the variable from "old rfc" to something more meaningful.